src/Components/Components/src/Routing/Router.cs

```diff
@@ -56,6 +61,16 @@ public class Router : IComponent, IHandleAfterRender, IDisposable
         /// </summary>
         [Parameter] public RenderFragment<RouteData> Found { get; set; }
 
+        /// <summary>
+        /// Get or sets the content to display when asynchronous navigation is in progress.
+        /// </summary>
+        [Parameter] public RenderFragment Navigating { get; set; }
+
+        /// <summary>
+        /// Gets or sets a handler that should be called before navigating to a new page.
+        /// </summary>
+        [Parameter] public Func<NavigationContext, Task> OnNavigateAsync { get; set; }
```

Comment:
I just realised: I think this should be an `EventCallback<NavigationContext>`. This will then behave as normal like other events when it comes to re-rendering the host component.

Reply:
@SteveSandersonMS good catch!

Reply:
Done.

Not pushed yet, but coming right up! 😛